csrc/predicate_compute.cpp

```diff
@@ -255,19 +255,53 @@ UnswitchPredicateKey::UnswitchPredicateKey()
 UnswitchPredicateKey::UnswitchPredicateKey(
     IterDomain* predicated_consumer_id,
     TensorView* consumer_tv,
-    IterDomain* predicated_concrete_id)
-    : predicated_concrete_id_(predicated_concrete_id) {
+    IterDomain* predicated_concrete_id,
+    std::unordered_set<IterDomain*> loop_ids)
+    : predicated_concrete_id_(predicated_concrete_id),
+      loop_ids_(std::move(loop_ids)) {
   // Initialize the parallelized domain map
+  // TODO: Add DID
   for (auto pt : kParallelTypeThreads) {
     parallel_concrete_ids_.insert({pt, nullptr});
   }
 
   std::vector<Val*> all_parallelized_consumer_loop_ids;
-  std::copy_if(
-      consumer_tv->getLoopDomain().begin(),
-      consumer_tv->getLoopDomain().end(),
-      std::back_inserter(all_parallelized_consumer_loop_ids),
-      [](IterDomain* x) { return isParallelTypeThread(x->getParallelType()); });
+
+  // Identify which parallel type is used for which loop domain for
```

Comment:
Added this fix. `UnswitchPredicateKey` was mainly designed for supporting shift predicates. I'll clean do some cleanup later.